main.odin

```diff
@@ -1077,160 +1077,129 @@ write_doc_line :: proc(w: io.Writer, text: string) {
-	s := s
-	n := strings.count(s, "`")
-	if n > 0 && n%2 == 0 {
-		for len(s) > 0 {
-			i := strings.index_byte(s, '`')
-			if i < 0 {
+write_markup_single_line :: proc(w: io.Writer, line: ^string) {
+	// Inline code
+	for {
+		backtick_index := strings.index_byte(line^, '`')
+		if backtick_index >= 0 {
+			io.write_string(w, line[:backtick_index])
+			line^ = line[backtick_index + 1:]
+
+			closing_backtick_index := strings.index_byte(line^, '`')
+			if closing_backtick_index >= 0 {
+				io.write_string(w, "<code>")
+				defer io.write_string(w, "</code>")
+				io.write_string(w, line[:closing_backtick_index])
+				line^ = line[closing_backtick_index + 1:]
+			} else {
 				break
 			}
-			io.write_string(w, s[:i])
-			s = s[i+1:]
-			io.write_string(w, "<code>")
-
-			i = strings.index_byte(s, '`'); assert(i >= 0)
-			io.write_string(w, s[:i])
-			s = s[i+1:]
-			io.write_string(w, "</code>")
+		} else {
+			break
 		}
-		io.write_string(w, s)
-	} else {
-		io.write_string(w, s)
-	}
-}
-
-write_docs :: proc(w: io.Writer, docs: string) {
-	if docs == "" {
-		return
-	}
-	Block_Kind :: enum {
-		Paragraph,
-		Code,
-	}
-	Block :: struct {
-		kind: Block_Kind,
-		lines: []string,
-	}
-
-	lines: [dynamic]string
-	it := docs
-	for line_ in strings.split_iterator(&it, "\n") {
-		line := strings.trim_right_space(line_)
-		append(&lines, line)
 	}
 
-	curr_block_kind := Block_Kind.Paragraph
-	start := 0
-	blocks: [dynamic]Block
-
-	for line, i in lines {
-		text := strings.trim_space(line)
-		switch curr_block_kind {
-		case .Paragraph:
-			if strings.has_prefix(line, "\t") {
-				if i-start > 0 {
-					append(&blocks, Block{curr_block_kind, lines[start:i]})
-				}
-				curr_block_kind, start = .Code, i
-			} else if text == "" {
-				if i-start > 0 {
-					append(&blocks, Block{curr_block_kind, lines[start:i]})
-				}
-				start = i
-			}
-		case .Code:
-			if text == "" || strings.has_prefix(line, "\t") {
-				continue
-			}
+	// Normal text
+	io.write_string(w, line^)
+	io.write_string(w, " ")
+}
 
-			if i-start > 0 {
-				append(&blocks, Block{curr_block_kind, lines[start:i]})
-			}
-			curr_block_kind, start = .Paragraph, i
-		}
-	}
-	if start < len(lines) {
-		if len(lines)-start > 0 {
-			append(&blocks, Block{curr_block_kind, lines[start:]})
+write_markup_text :: proc(w: io.Writer, lines: []string) {
+	io.write_string(w, "<span style=\"white-space: pre-wrap\">") // Preserve leading whitespace
+	io.write_string(w, "<p>")
+	
+	line_loop: for i := 0; i < len(lines); i += 1 {
+		line := lines[i]
+		trimmed_line := strings.trim_space(line)
+
+		// Paragraph break
+		if len(trimmed_line) == 0 {
+			io.write_string(w, "</p>")
+			io.write_string(w, "<p>")
+			continue
 		}
-	}
 
-	for block in &blocks {
-		trim_amount := 0
-		for trim_amount = 0; trim_amount < len(block.lines); trim_amount += 1 {
-			line := block.lines[trim_amount]
-			if strings.trim_space(line) != "" {
-				break
+		// Lists
+		if trimmed_line[0] == '-' {
```

Comment:
I would allow for both `-` and `*`.

I personally prefer `*` over `-` for lists in markdown-like languages.